### DISCUSSION TITLE
Summary-based information flow analysis for javascript

### DISCUSSION THREAD

**Discussion by nettrino** [github/codeql#10483] [Q&A]
Hello all, 

I am trying to replicate the steps in the (rather old) [javascript doc for information flow analysis](https://github.com/github/codeql/blob/main/javascript/documentation/flow-summaries.rst). In particular I have done the following:

1. Checked out [node-mkdir](https://github.com/isaacs/node-mkdirp)
2. Created a database from within the root of the repo using `codeql database create ~/.local/codeql/databases/js-mkdir --language=javascript --overwrite`
3. Run an analysis using `codeql query run --database ~/.local/codeql/databases/js-mkdir --threads=8 --ram=10000 -- qlstarter/ql/javascript/ql/src/experimental/Summaries/ExtractSourceSummaries.ql`

I cannot see any of the results outlined in the documentation, which has references to older Semmle tooling. How can I achieve equivalent results as described in the docs using the latest codeql? 

P.S. I also added some comment directives in the repo's `/lib/mkdirp-native.js` without much luck:

```diff
 const mkdirpNative = (path, opts) => {
   opts.recursive = true
-  const parent = dirname(path)
+  const parent = dirname(
+    path /* Semmle: sink: taint, TaintedPath */
+  )
```

I do see how the intent of the doc is to examine/analyze applications actually using `mkdirp` and not the package itself, however I would expect there would be source summaries just by analysing mkdirp as a standalone pkg?

**Comment by nettrino** [comment]
I  _think_ I understood better what the issue is after going through https://codeql.github.com/docs/codeql-cli/manual/database-analyze/. 

The results of the query are generated if I run
`codeql database run-queries --threads=8 --ram=10000 -v -- ~/.local/codeql/databases/js-mkdir qlstarter/ql/javascript/ql/src/experimental/Summaries/ExtractSourceSummaries.ql`, and stored locally in the database folder, however they are still empty if inspected via

`codeql bqrs decode ~/.local/codeql/databases/js-mkdir/results/codeql/javascript-queries/experimental/Summaries/ExtractSourceSummaries.bqrs`

Am I missing something?

**Comment by asgerf** [comment]
Hi @nettrino,

Sorry but the document is so far out of date that I'd say it's not worth trying to make it work again. I am going to remove the document and the queries, as it describes a feature that is unsupported and unmaintained. Apologies for any inconvenience this has caused you.

**Reply by nettrino** [reply]
No problem - and thank you for the reply! I am interested in the functionality overall so if there is a way to achieve the equivalent of extracting such summaries with the current APIs that would be great!
